src/components/trends/common/Tooltip.js

```diff
@@ -0,0 +1,184 @@
+          break;
+        case 'left':
+        default:
+          leftOffset = -tooltipRect.width;
+          topOffset = -tooltipRect.height / 2;
+      }
+      offset.top = topOffset + propOffset.top;
+      offset.left = leftOffset + propOffset.left;
+    }
+
+    this.setState({ offset });
+  }
+
+  renderTail(color = 'white') {
+    const { tailWidth, tailHeight, borderWidth, borderColor, side } = this.props;
+    const tailSide = side === 'left' ? 'right' : 'left';
+    const padding = 10;
+    let marginOuterValue;
+    let marginInnerValue;
+    if (tailSide === 'left') {
+      marginOuterValue = `calc(-100% - 2 * ${padding}px)`;
+      marginInnerValue = `calc(-100% - (2 * ${padding}px - ${borderWidth + 1}px))`;
+    } else {
+      marginOuterValue = `calc(${padding}px + ${borderWidth}px)`;
+      marginInnerValue = `${padding - 1}px`;
+    }
+    const borderSide = tailSide === 'left' ? 'right' : 'left';
+
+    return (<div>
+      <div
+        ref={(ref) => { this.tailElem = ref; }}
+        className={styles.tail}
+        style={{
+          marginTop: `-${tailHeight}px`,
+          marginLeft: marginOuterValue,
+          borderWidth: `${tailHeight}px ${2 * tailWidth}px`,
+          [`border${_.capitalize(borderSide)}Color`]: borderColor,
+        }}
+      ></div>
+      <div
+        className={styles.tail}
+        style={{
+          marginTop: `-${tailHeight}px`,
+          marginLeft: marginInnerValue,
+          borderWidth: `${tailHeight}px ${2 * tailWidth}px`,
+          [`border${_.capitalize(borderSide)}Color`]: color,
+        }}
+      ></div>
+    </div>);
+  }
+
+  renderTitle(title) {
+    const { tail, content } = this.props;
+    let renderedTitle = null;
+    if (title) {
+      renderedTitle = (<div className={styles.title}>
+        <span>{title}</span>
+        {tail && !content && this.renderTail('#EAEAEE')}
+      </div>);
+    }
+    return renderedTitle;
+  }
+
+  renderContent(content) {
+    let renderedContent = null;
+    const { tail } = this.props;
+    if (content) {
+      renderedContent = (<div className={styles.content}>
+        <span>{content}</span>
+        {tail && this.renderTail()}
+      </div>);
+    }
+    return renderedContent;
+  }
+
+  render() {
+    const { title, content, position, borderColor, borderWidth } = this.props;
+    const { offset } = this.state;
+    const top = position.top + offset.top;
+    const left = position.left + offset.left;
+
+    return (
+      <div
+        className={styles.tooltip}
+        style={{ top, left, borderColor, borderWidth: `${borderWidth}px` }}
+        ref={(ref) => { this.element = ref; }}
+      >
+        {title && this.renderTitle(title)}
+        {content && this.renderContent(content)}
+      </div>
+    );
+  }
+}
+
+Tooltip.propTypes = {
+  title: PropTypes.node,
+  content: PropTypes.node,
+  position: PropTypes.shape({
+    top: PropTypes.number.isRequired,
+    left: PropTypes.number.isRequired,
+  }),
```

Comment:
I think what you've just described is that it's not required for the component to not blow up, but it is pretty much required for how we expect the component to be used. So I think `isRequired` would be helpful, cuz then it gives users of the component React warning if they do it wrong.